src/ui/ui.py

```diff
@@ -1266,8 +1266,38 @@ def generate_report_and_download(report_format: str):
                     summarize_model_id=multi_model_name,
                     api_key=api_key,
                 )
-                # Parse minimal fields back from MCP text
-                result = parse_analyze_response(result_text) if result_text else {}
+
+                # Prefer client-side structured error (dict format) so we can render cleanly
+                if isinstance(result, dict) and "error" in result:
```

Comment:
is that possible to move this into error handler file ?

Reply:
Good idea. Done!